### PR TITLE
2191 destroy dup legends

### PR DESCRIPTION
**Pull request by amartinj**
Closes https://github.com/CartoDB/cartodb-central/issues/2191

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -321,6 +321,7 @@ SPEC_HELPER_MIN_SPECS = \
 	spec/lib/carto/storage_options/local_spec.rb \
 	spec/lib/carto/visualization_invalidation_service_spec.rb \
 	spec/lib/tasks/layers_rake_spec.rb \
+	spec/lib/tasks/fix_unique_legends.rake \
```
Comment (javitonino) [inline Makefile]: :eyes:

File: lib/tasks/fix_unique_legends.rake [hunk 2]
```diff
@@ -0,0 +1,10 @@
+namespace :cartodb do
+  desc "fixes duplicated unique legends"
+  task fix_unique_legends: :environment do
+    query = 'SELECT layer_id, count(*) from legends group by layer_id having count(*) > 1'
+    ActiveRecord::Base.connection.execute(query).each do |row|
+      legends = Carto::Layer.find(row['layer_id']).legends
+      legends.sort_by(&:updated_at).slice(1..legends.count).each(&:destroy)
```
Comment (javitonino) [inline lib/tasks/fix_unique_legends.rake]: This only leaves the most recent layer. It needs to leave the most recent of each category (size or color).

File: spec/lib/tasks/fix_unique_legends_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @layer.reload.legends.first.id.should eq @legend.id
+  end
+
+  it 'keeps layers if different type' do
+    legend2 = @legend.dup
+    legend2.type = 'bubble'
+    legend2.save(validate: false)
+    @layer.reload.legends.count.should eq 2
+
+    Rake.application['cartodb:fix_unique_legends'].invoke
```
Comment (javitonino) [inline spec/lib/tasks/fix_unique_legends_spec.rb]: This is not executed. You need to reenable the rake or it only runs once. eg: `Rake.application['cartodb:fix_unique_legends'].reenable`

Rakes are singletons and will only be invoked once in a process lifetime unless explicitly told to be reenabled.

**Comment by javitonino** [comment]
Acceptance :green_apple:

Just fix that test problem, and ready to go.

**Comment by amartinj** [comment]
done

**Comment by javitonino** [comment]
retest this please

**Comment by javitonino** [comment]
Merged and rake executed

**Pull request merged**